flocker/node/agents/test/test_blockdevice.py

```diff
@@ -1722,6 +1723,28 @@ def test_get_device_path_unattached_volume(self):
         )
         self.assertEqual(new_volume.blockdevice_id, exception.blockdevice_id)
 
+    def test_get_device_path_detached_volume(self):
+        """
+        ``get_device_path`` raises ``UnattachedVolume`` if the supplied
+        ``blockdevice_id`` corresponds to a volume that was attached to the
+        node but has been detached from it.
+        """
```

Comment:
In old implementation EC2 APIs were queried each time. So it was same code path as never seen before unattached volume. In new implementation we can only give information about volumes that have been attached during the lifetime of the process, so this particular test covers a new code path in the new implementation.